src/spanningtrees/prim.jl

```diff
@@ -0,0 +1,64 @@
+immutable PrimHeapEntry{T<:Real}
+    edge::Edge
+    dist::T
+end
+
+isless(e1::PrimHeapEntry, e2::PrimHeapEntry) = e1.dist < e2.dist
+
+"""Performs [Prim's algorithm](https://en.wikipedia.org/wiki/Prim%27s_algorithm)
+on a connected, non-directional graph `g`, having adjacency matrix `distmx`,
+and computes minimum spanning tree. Returns a `Vector{Edge}`,
+that contains the edges.
+"""
+function prim_mst{T<:Real}(
+    g::SimpleGraph,
+    distmx::AbstractArray{T, 2} = DefaultDistance()
+)
+    pq = Vector{PrimHeapEntry{T}}()
+    mst = Vector{Edge}()
+    marked = zeros(Bool, nv(g))
+
+    sizehint!(pq, ne(g))
+    sizehint!(mst, ne(g))
+    visit(g, 1, marked, pq, distmx)
+
+    while !isempty(pq)
+        heap_entry = heappop!(pq)
+        v = src(heap_entry.edge)
+        w = dst(heap_entry.edge)
+
+        if marked[v] && marked[w]
+            continue
+        end
+        push!(mst, heap_entry.edge)
+
+        if !marked[v]
+            visit(g, v, marked, pq, distmx)
+        end
+
+        if !marked[w]
+            visit(g, w, marked, pq, distmx)
+        end
+    end
+
+    return mst
+end
+
+"""
+Used to mark the visited vertices. Marks the vertex `v` of graph `g` true in the array `marked`
+and enters all its edges into priority queue `pq` with its `distmx` values as a PrimHeapEntry.
+"""
+function visit{T<:Real}(
```

Comment:
Since this is mutating one of its arguments, I think the name should end in `!`.